src/install-ocaml-windows.cmd

```diff
@@ -1,7 +1,43 @@
+@setlocal
+@echo off
+
+:: This must be the /cygdrive-relative version of CYGWIN_ROOT in constants.ts
+set CYGWIN_NATIVE_ROOT=/cygdrive/d/cygwin
+
+set CACHE_ROOT=%GITHUB_WORKSPACE%\.github\caches
+set OPAM_CACHE=%CACHE_ROOT%\opam\cache.tar
+
+if not exist %CACHE_ROOT%\opam\bootstrap\nul md %CACHE_ROOT%\opam\bootstrap
+
+if exist %OPAM_CACHE% (
+  set INITIALISING=0
+  if not exist %CYGWIN_ROOT%\bin\nul md %CYGWIN_ROOT%\bin
+  copy %CACHE_ROOT%\opam\bootstrap\* %CYGWIN_ROOT%\bin\
+  cd %CACHE_ROOT%\opam
+  %CYGWIN_ROOT%\bin\tar -pxf cache.tar -C "%CYGWIN_NATIVE_ROOT%"
```

Comment:
Bootstrapping works by putting `tar.exe` (and its dependent DLLs) in `%CYGWIN_ROOT%\bin\`. When that `tar.exe` is run, `cygwin1.dll` "boots" and will recognise `%CYGWIN_ROOT%` as `/` (by design - that's how the DLL works). The tarball is then untarred - not to `/` (which would fail for the `/dev` symlinks) but to `/cygdrive/d/cygwin` - i.e. to the mount. This works largely like a `dd` directly to the drive and restores the entire Cygwin installation... critically it also restores the junction points correctly.